src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2942,7 +3050,20 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 visit_ClassDef(*AST::down_cast<AST::ClassDef_t>(x.m_body[i]));
                 continue;
             } else if ( AST::is_a<AST::FunctionDef_t>(*x.m_body[i]) ) {
-                throw SemanticError("Struct member functions are not supported", x.m_body[i]->base.loc);
+                if ( !is_class_scope ) {
+                    throw SemanticError("Struct member functions are not supported", x.m_body[i]->base.loc);
+                }
+                else {
+                    AST::FunctionDef_t *f = AST::down_cast<AST::FunctionDef_t>(x.m_body[i]);
+                    member_fn_names.push_back(al, f->m_name);
+                    std::string f_name = f->m_name;
+                    if (f_name == "__init__"){
+                        this->visit_Constructor(*f,member_names,member_init);
```

Comment:
Minor: How about?
```suggestion
                        this->visit_initalizer_method(*f,member_names,member_init);
```
or 
```suggestion
                        this->visit_initalizer_function(*f,member_names,member_init);
```
or
```suggestion
                        this->handle_init_method(*f,member_names,member_init);
```